{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "4c78195d",
   "metadata": {},
   "source": [
    "# CMRG HG002: overlap of small variants by ClinVar CLNSIG category with homopolymer regions\n",
    "\n",
    "This notebook generates Table 1 and 2 of the manuscript, with the ClinVar SNVs and Indels that overlap with homopolymer regions.\n",
    "The input file `clinvar_overlap_by_CLNSIG_and_VAR_TYPE.csv` is generated with the R script `analyses/clinvar_overlap/find_overlap.R`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "412a8b56",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# adjust the path to the input file accordingly\n",
    "clinvar_res = \"clinvar_overlap_by_CLNSIG_and_VAR_TYPE.csv\"\n",
    "\n",
    "df = pd.read_csv(clinvar_res, sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "4f5d4bf3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BED_file</th>\n",
       "      <th>VAR_TYPE</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Overlap_variants</th>\n",
       "      <th>Non_overlap_variants</th>\n",
       "      <th>Total_variants</th>\n",
       "      <th>Overlap_proportion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GRCh38_SimpleRepeat_homopolymer_4to6_slop5</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Affects</td>\n",
       "      <td>21</td>\n",
       "      <td>85</td>\n",
       "      <td>106</td>\n",
       "      <td>0.1981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GRCh38_SimpleRepeat_homopolymer_4to6_slop5</td>\n",
       "      <td>SNP</td>\n",
       "      <td>association</td>\n",
       "      <td>74</td>\n",
       "      <td>222</td>\n",
       "      <td>296</td>\n",
       "      <td>0.2500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GRCh38_SimpleRepeat_homopolymer_4to6_slop5</td>\n",
       "      <td>SNP</td>\n",
       "      <td>association_not_found</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRCh38_SimpleRepeat_homopolymer_4to6_slop5</td>\n",
       "      <td>SNP</td>\n",
       "      <td>association|drug_response|risk_factor</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRCh38_SimpleRepeat_homopolymer_4to6_slop5</td>\n",
       "      <td>SNP</td>\n",
       "      <td>association|risk_factor</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     BED_file VAR_TYPE  \\\n",
       "0  GRCh38_SimpleRepeat_homopolymer_4to6_slop5      SNP   \n",
       "1  GRCh38_SimpleRepeat_homopolymer_4to6_slop5      SNP   \n",
       "2  GRCh38_SimpleRepeat_homopolymer_4to6_slop5      SNP   \n",
       "3  GRCh38_SimpleRepeat_homopolymer_4to6_slop5      SNP   \n",
       "4  GRCh38_SimpleRepeat_homopolymer_4to6_slop5      SNP   \n",
       "\n",
       "                                  CLNSIG  Overlap_variants  \\\n",
       "0                                Affects                21   \n",
       "1                            association                74   \n",
       "2                  association_not_found                 0   \n",
       "3  association|drug_response|risk_factor                 0   \n",
       "4                association|risk_factor                 0   \n",
       "\n",
       "   Non_overlap_variants  Total_variants  Overlap_proportion  \n",
       "0                    85             106              0.1981  \n",
       "1                   222             296              0.2500  \n",
       "2                     4               4              0.0000  \n",
       "3                     1               1              0.0000  \n",
       "4                     1               1              0.0000  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "9d7668fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BED_file</th>\n",
       "      <th>VAR_TYPE</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Overlap_variants</th>\n",
       "      <th>Non_overlap_variants</th>\n",
       "      <th>Total_variants</th>\n",
       "      <th>Overlap_proportion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Affects</td>\n",
       "      <td>21</td>\n",
       "      <td>85</td>\n",
       "      <td>106</td>\n",
       "      <td>0.1981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>association</td>\n",
       "      <td>74</td>\n",
       "      <td>222</td>\n",
       "      <td>296</td>\n",
       "      <td>0.2500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>association_not_found</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>association|drug_response|risk_factor</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>association|risk_factor</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    BED_file VAR_TYPE                                 CLNSIG  \\\n",
       "0  hpol_4to6      SNP                                Affects   \n",
       "1  hpol_4to6      SNP                            association   \n",
       "2  hpol_4to6      SNP                  association_not_found   \n",
       "3  hpol_4to6      SNP  association|drug_response|risk_factor   \n",
       "4  hpol_4to6      SNP                association|risk_factor   \n",
       "\n",
       "   Overlap_variants  Non_overlap_variants  Total_variants  Overlap_proportion  \n",
       "0                21                    85             106              0.1981  \n",
       "1                74                   222             296              0.2500  \n",
       "2                 0                     4               4              0.0000  \n",
       "3                 0                     1               1              0.0000  \n",
       "4                 0                     1               1              0.0000  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_dict = {\n",
    "    'GRCh38_SimpleRepeat_homopolymer_4to6_slop5': 'hpol_4to6',\n",
    "    'GRCh38_SimpleRepeat_homopolymer_7to11_slop5': 'hpol_7to11',\n",
    "    'GRCh38_SimpleRepeat_homopolymer_ge12_slop5': 'hpol_ge12',\n",
    "    'GRCh38_SimpleRepeat_homopolymer_ge21_slop5': 'hpol_ge21'\n",
    "}\n",
    "\n",
    "df['BED_file'] = df['BED_file'].replace(subset_dict)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "1b90e419",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Affects', 'association', 'association_not_found',\n",
       "       'association|drug_response|risk_factor', 'association|risk_factor',\n",
       "       'Benign', 'Benign/Likely_benign',\n",
       "       'Benign/Likely_benign|association',\n",
       "       'Benign/Likely_benign|drug_response',\n",
       "       'Benign/Likely_benign|drug_response|other',\n",
       "       'Benign/Likely_benign|other',\n",
       "       'Benign/Likely_benign|other|risk_factor',\n",
       "       'Benign/Likely_benign|risk_factor', 'Benign|Affects',\n",
       "       'Benign|Affects|association|other', 'Benign|association',\n",
       "       'Benign|confers_sensitivity', 'Benign|drug_response',\n",
       "       'Benign|other', 'Benign|protective', 'Benign|risk_factor',\n",
       "       'confers_sensitivity', 'confers_sensitivity|other',\n",
       "       'Conflicting_classifications_of_pathogenicity',\n",
       "       'Conflicting_classifications_of_pathogenicity|Affects',\n",
       "       'Conflicting_classifications_of_pathogenicity|association',\n",
       "       'Conflicting_classifications_of_pathogenicity|association|risk_factor',\n",
       "       'Conflicting_classifications_of_pathogenicity|drug_response',\n",
       "       'Conflicting_classifications_of_pathogenicity|other',\n",
       "       'Conflicting_classifications_of_pathogenicity|other|risk_factor',\n",
       "       'Conflicting_classifications_of_pathogenicity|protective',\n",
       "       'Conflicting_classifications_of_pathogenicity|risk_factor',\n",
       "       'drug_response', 'drug_response|other', 'Established_risk_allele',\n",
       "       'Likely_benign', 'Likely_benign|Affects|association',\n",
       "       'Likely_benign|association', 'Likely_benign|drug_response',\n",
       "       'Likely_benign|drug_response|other', 'Likely_benign|other',\n",
       "       'Likely_benign|risk_factor', 'Likely_pathogenic',\n",
       "       'Likely_pathogenic,_low_penetrance',\n",
       "       'Likely_pathogenic/Likely_pathogenic,_low_penetrance',\n",
       "       'Likely_pathogenic/Likely_risk_allele',\n",
       "       'Likely_pathogenic/Pathogenic,_low_penetrance',\n",
       "       'Likely_pathogenic|Affects', 'Likely_pathogenic|association',\n",
       "       'Likely_pathogenic|drug_response', 'Likely_pathogenic|protective',\n",
       "       'Likely_pathogenic|risk_factor', 'Likely_risk_allele', nan,\n",
       "       'no_classification_for_the_single_variant',\n",
       "       'no_classifications_from_unflagged_records', 'not_provided',\n",
       "       'other', 'other|risk_factor', 'Pathogenic',\n",
       "       'Pathogenic,_low_penetrance', 'Pathogenic/Likely_pathogenic',\n",
       "       'Pathogenic/Likely_pathogenic,_low_penetrance',\n",
       "       'Pathogenic/Likely_pathogenic/Likely_risk_allele',\n",
       "       'Pathogenic/Likely_pathogenic/Pathogenic,_low_penetrance',\n",
       "       'Pathogenic/Likely_pathogenic/Pathogenic,_low_penetrance|risk_factor',\n",
       "       'Pathogenic/Likely_pathogenic|association',\n",
       "       'Pathogenic/Likely_pathogenic|other',\n",
       "       'Pathogenic/Likely_pathogenic|risk_factor',\n",
       "       'Pathogenic/Likely_risk_allele',\n",
       "       'Pathogenic/Pathogenic,_low_penetrance',\n",
       "       'Pathogenic/Pathogenic,_low_penetrance|other',\n",
       "       'Pathogenic|Affects', 'Pathogenic|association',\n",
       "       'Pathogenic|confers_sensitivity', 'Pathogenic|drug_response',\n",
       "       'Pathogenic|other', 'Pathogenic|risk_factor', 'protective',\n",
       "       'protective|risk_factor', 'risk_factor', 'Uncertain_risk_allele',\n",
       "       'Uncertain_risk_allele|risk_factor', 'Uncertain_significance',\n",
       "       'Uncertain_significance/Uncertain_risk_allele',\n",
       "       'Uncertain_significance|Affects',\n",
       "       'Uncertain_significance|association',\n",
       "       'Uncertain_significance|drug_response',\n",
       "       'Uncertain_significance|other',\n",
       "       'Uncertain_significance|risk_factor',\n",
       "       'Conflicting_classifications_of_pathogenicity|drug_response|other',\n",
       "       'drug_response|risk_factor', 'Pathogenic|protective'], dtype=object)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['CLNSIG'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "3a5d9aaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "clnsig_subset = [\n",
    "    'Benign',\n",
    "    'Likely_benign',\n",
    "    'Uncertain_significance',\n",
    "    'Likely_pathogenic',\n",
    "    'Pathogenic'\n",
    "]\n",
    "\n",
    "df = df[df['CLNSIG'].isin(clnsig_subset)]\n",
    "df['CLNSIG'] = pd.Categorical(df['CLNSIG'], categories=clnsig_subset, ordered=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "51917637",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BED_file</th>\n",
       "      <th>VAR_TYPE</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Overlap_variants</th>\n",
       "      <th>Non_overlap_variants</th>\n",
       "      <th>Total_variants</th>\n",
       "      <th>Overlap_proportion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [BED_file, VAR_TYPE, CLNSIG, Overlap_variants, Non_overlap_variants, Total_variants, Overlap_proportion]\n",
       "Index: []"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['Non_overlap_variants']<0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "c5003bb0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BED_file</th>\n",
       "      <th>VAR_TYPE</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Overlap_variants</th>\n",
       "      <th>Non_overlap_variants</th>\n",
       "      <th>Total_variants</th>\n",
       "      <th>Overlap_proportion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Benign</td>\n",
       "      <td>48677</td>\n",
       "      <td>131569</td>\n",
       "      <td>180246</td>\n",
       "      <td>0.2701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>261084</td>\n",
       "      <td>710021</td>\n",
       "      <td>971105</td>\n",
       "      <td>0.2689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>15228</td>\n",
       "      <td>55628</td>\n",
       "      <td>70856</td>\n",
       "      <td>0.2149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>16696</td>\n",
       "      <td>65333</td>\n",
       "      <td>82029</td>\n",
       "      <td>0.2035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>458592</td>\n",
       "      <td>1600725</td>\n",
       "      <td>2059317</td>\n",
       "      <td>0.2227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Benign</td>\n",
       "      <td>10957</td>\n",
       "      <td>17498</td>\n",
       "      <td>28455</td>\n",
       "      <td>0.3851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>14298</td>\n",
       "      <td>22876</td>\n",
       "      <td>37174</td>\n",
       "      <td>0.3846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>10959</td>\n",
       "      <td>24325</td>\n",
       "      <td>35284</td>\n",
       "      <td>0.3106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>28537</td>\n",
       "      <td>63149</td>\n",
       "      <td>91686</td>\n",
       "      <td>0.3112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>hpol_4to6</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>16926</td>\n",
       "      <td>41668</td>\n",
       "      <td>58594</td>\n",
       "      <td>0.2889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Benign</td>\n",
       "      <td>3116</td>\n",
       "      <td>177130</td>\n",
       "      <td>180246</td>\n",
       "      <td>0.0173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>12259</td>\n",
       "      <td>958846</td>\n",
       "      <td>971105</td>\n",
       "      <td>0.0126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>435</td>\n",
       "      <td>70421</td>\n",
       "      <td>70856</td>\n",
       "      <td>0.0061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>377</td>\n",
       "      <td>81652</td>\n",
       "      <td>82029</td>\n",
       "      <td>0.0046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>10981</td>\n",
       "      <td>2048336</td>\n",
       "      <td>2059317</td>\n",
       "      <td>0.0053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Benign</td>\n",
       "      <td>5410</td>\n",
       "      <td>23045</td>\n",
       "      <td>28455</td>\n",
       "      <td>0.1901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>3404</td>\n",
       "      <td>33770</td>\n",
       "      <td>37174</td>\n",
       "      <td>0.0916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>391</td>\n",
       "      <td>34893</td>\n",
       "      <td>35284</td>\n",
       "      <td>0.0111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1443</td>\n",
       "      <td>90243</td>\n",
       "      <td>91686</td>\n",
       "      <td>0.0157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>hpol_7to11</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>1481</td>\n",
       "      <td>57113</td>\n",
       "      <td>58594</td>\n",
       "      <td>0.0253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Benign</td>\n",
       "      <td>1245</td>\n",
       "      <td>179001</td>\n",
       "      <td>180246</td>\n",
       "      <td>0.0069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>2201</td>\n",
       "      <td>968904</td>\n",
       "      <td>971105</td>\n",
       "      <td>0.0023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>86</td>\n",
       "      <td>70770</td>\n",
       "      <td>70856</td>\n",
       "      <td>0.0012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>69</td>\n",
       "      <td>81960</td>\n",
       "      <td>82029</td>\n",
       "      <td>0.0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>758</td>\n",
       "      <td>2058559</td>\n",
       "      <td>2059317</td>\n",
       "      <td>0.0004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Benign</td>\n",
       "      <td>6049</td>\n",
       "      <td>22406</td>\n",
       "      <td>28455</td>\n",
       "      <td>0.2126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>3469</td>\n",
       "      <td>33705</td>\n",
       "      <td>37174</td>\n",
       "      <td>0.0933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>37</td>\n",
       "      <td>35247</td>\n",
       "      <td>35284</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>35</td>\n",
       "      <td>91651</td>\n",
       "      <td>91686</td>\n",
       "      <td>0.0004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464</th>\n",
       "      <td>hpol_ge12</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>1184</td>\n",
       "      <td>57410</td>\n",
       "      <td>58594</td>\n",
       "      <td>0.0202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Benign</td>\n",
       "      <td>291</td>\n",
       "      <td>179955</td>\n",
       "      <td>180246</td>\n",
       "      <td>0.0016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>524</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>401</td>\n",
       "      <td>970704</td>\n",
       "      <td>971105</td>\n",
       "      <td>0.0004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>10</td>\n",
       "      <td>70846</td>\n",
       "      <td>70856</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>16</td>\n",
       "      <td>82013</td>\n",
       "      <td>82029</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>160</td>\n",
       "      <td>2059157</td>\n",
       "      <td>2059317</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Benign</td>\n",
       "      <td>1262</td>\n",
       "      <td>27193</td>\n",
       "      <td>28455</td>\n",
       "      <td>0.0444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>911</td>\n",
       "      <td>36263</td>\n",
       "      <td>37174</td>\n",
       "      <td>0.0245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>35271</td>\n",
       "      <td>35284</td>\n",
       "      <td>0.0004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>612</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>9</td>\n",
       "      <td>91677</td>\n",
       "      <td>91686</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>627</th>\n",
       "      <td>hpol_ge21</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>342</td>\n",
       "      <td>58252</td>\n",
       "      <td>58594</td>\n",
       "      <td>0.0058</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       BED_file VAR_TYPE                  CLNSIG  Overlap_variants  \\\n",
       "5     hpol_4to6      SNP                  Benign             48677   \n",
       "35    hpol_4to6      SNP           Likely_benign            261084   \n",
       "42    hpol_4to6      SNP       Likely_pathogenic             15228   \n",
       "59    hpol_4to6      SNP              Pathogenic             16696   \n",
       "83    hpol_4to6      SNP  Uncertain_significance            458592   \n",
       "92    hpol_4to6    INDEL                  Benign             10957   \n",
       "107   hpol_4to6    INDEL           Likely_benign             14298   \n",
       "110   hpol_4to6    INDEL       Likely_pathogenic             10959   \n",
       "123   hpol_4to6    INDEL              Pathogenic             28537   \n",
       "138   hpol_4to6    INDEL  Uncertain_significance             16926   \n",
       "168  hpol_7to11      SNP                  Benign              3116   \n",
       "198  hpol_7to11      SNP           Likely_benign             12259   \n",
       "205  hpol_7to11      SNP       Likely_pathogenic               435   \n",
       "222  hpol_7to11      SNP              Pathogenic               377   \n",
       "246  hpol_7to11      SNP  Uncertain_significance             10981   \n",
       "255  hpol_7to11    INDEL                  Benign              5410   \n",
       "270  hpol_7to11    INDEL           Likely_benign              3404   \n",
       "273  hpol_7to11    INDEL       Likely_pathogenic               391   \n",
       "286  hpol_7to11    INDEL              Pathogenic              1443   \n",
       "301  hpol_7to11    INDEL  Uncertain_significance              1481   \n",
       "331   hpol_ge12      SNP                  Benign              1245   \n",
       "361   hpol_ge12      SNP           Likely_benign              2201   \n",
       "368   hpol_ge12      SNP       Likely_pathogenic                86   \n",
       "385   hpol_ge12      SNP              Pathogenic                69   \n",
       "409   hpol_ge12      SNP  Uncertain_significance               758   \n",
       "418   hpol_ge12    INDEL                  Benign              6049   \n",
       "433   hpol_ge12    INDEL           Likely_benign              3469   \n",
       "436   hpol_ge12    INDEL       Likely_pathogenic                37   \n",
       "449   hpol_ge12    INDEL              Pathogenic                35   \n",
       "464   hpol_ge12    INDEL  Uncertain_significance              1184   \n",
       "494   hpol_ge21      SNP                  Benign               291   \n",
       "524   hpol_ge21      SNP           Likely_benign               401   \n",
       "531   hpol_ge21      SNP       Likely_pathogenic                10   \n",
       "548   hpol_ge21      SNP              Pathogenic                16   \n",
       "572   hpol_ge21      SNP  Uncertain_significance               160   \n",
       "581   hpol_ge21    INDEL                  Benign              1262   \n",
       "596   hpol_ge21    INDEL           Likely_benign               911   \n",
       "599   hpol_ge21    INDEL       Likely_pathogenic                13   \n",
       "612   hpol_ge21    INDEL              Pathogenic                 9   \n",
       "627   hpol_ge21    INDEL  Uncertain_significance               342   \n",
       "\n",
       "     Non_overlap_variants  Total_variants  Overlap_proportion  \n",
       "5                  131569          180246              0.2701  \n",
       "35                 710021          971105              0.2689  \n",
       "42                  55628           70856              0.2149  \n",
       "59                  65333           82029              0.2035  \n",
       "83                1600725         2059317              0.2227  \n",
       "92                  17498           28455              0.3851  \n",
       "107                 22876           37174              0.3846  \n",
       "110                 24325           35284              0.3106  \n",
       "123                 63149           91686              0.3112  \n",
       "138                 41668           58594              0.2889  \n",
       "168                177130          180246              0.0173  \n",
       "198                958846          971105              0.0126  \n",
       "205                 70421           70856              0.0061  \n",
       "222                 81652           82029              0.0046  \n",
       "246               2048336         2059317              0.0053  \n",
       "255                 23045           28455              0.1901  \n",
       "270                 33770           37174              0.0916  \n",
       "273                 34893           35284              0.0111  \n",
       "286                 90243           91686              0.0157  \n",
       "301                 57113           58594              0.0253  \n",
       "331                179001          180246              0.0069  \n",
       "361                968904          971105              0.0023  \n",
       "368                 70770           70856              0.0012  \n",
       "385                 81960           82029              0.0008  \n",
       "409               2058559         2059317              0.0004  \n",
       "418                 22406           28455              0.2126  \n",
       "433                 33705           37174              0.0933  \n",
       "436                 35247           35284              0.0010  \n",
       "449                 91651           91686              0.0004  \n",
       "464                 57410           58594              0.0202  \n",
       "494                179955          180246              0.0016  \n",
       "524                970704          971105              0.0004  \n",
       "531                 70846           70856              0.0001  \n",
       "548                 82013           82029              0.0002  \n",
       "572               2059157         2059317              0.0001  \n",
       "581                 27193           28455              0.0444  \n",
       "596                 36263           37174              0.0245  \n",
       "599                 35271           35284              0.0004  \n",
       "612                 91677           91686              0.0001  \n",
       "627                 58252           58594              0.0058  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df[df['VAR_TYPE'].isin(['SNP','INDEL'])]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "41a50634",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_snp = df[df['VAR_TYPE']=='SNP'][['BED_file','CLNSIG','Overlap_variants','Total_variants','Overlap_proportion']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "3167d50a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepare_df(df, vartype=\"SNP\"):\n",
    "    \n",
    "    subset_dict = {\n",
    "        'hpol_4to6': '4-6',\n",
    "        'hpol_7to11': '7-11',\n",
    "        'hpol_ge12': '>11',\n",
    "        'hpol_ge21': '>20'\n",
    "    }\n",
    "\n",
    "    df_res = df[df['VAR_TYPE']==vartype][['BED_file','CLNSIG','Overlap_variants','Total_variants','Overlap_proportion']].copy()\n",
    "\n",
    "    df_res['BED_file'] = df_res['BED_file'].replace(subset_dict)\n",
    "\n",
    "    df_res = df_res.rename(columns={\"BED_file\": \"HPol (bp)\", \"Overlap_variants\": \"Overlap\",\n",
    "                                    \"Total_variants\": \"Total\", \"Overlap_proportion\": \"%\"})\n",
    "    \n",
    "    return df_res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1a529e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from io import StringIO\n",
    "import re\n",
    "\n",
    "def latex_escape(s):\n",
    "    \"\"\"Escape LaTeX special characters (_ % & $ # ^ { } ~ \\\\)\"\"\"\n",
    "    if pd.isna(s):\n",
    "        return \"\"\n",
    "    s = str(s)\n",
    "    # Wrap strings containing '>' in math mode\n",
    "    if '>' in s:\n",
    "        # Escape underscores inside math mode\n",
    "        s = re.sub(r'_', r'\\_', s)\n",
    "        return f\"${s}$\"\n",
    "    # Otherwise escape normally\n",
    "    return re.sub(r'([_%&#$^{}~\\\\])', r'\\\\\\1', s)\n",
    "\n",
    "def df_to_latex_multirow(\n",
    "    df,\n",
    "    group_cols,\n",
    "    column_format=None,\n",
    "    caption=None,\n",
    "    label=None,\n",
    "    center=True,\n",
    "):\n",
    "    ncols = len(df.columns)\n",
    "    if column_format is None:\n",
    "        column_format = \"\".join([\"c\"] * ncols)\n",
    "\n",
    "    buf = StringIO()\n",
    "    if center:\n",
    "        buf.write(\"\\\\begin{table}[ht]\\n\\\\centering\\n\")\n",
    "    else:\n",
    "        buf.write(\"\\\\begin{table}[ht]\\n\")\n",
    "\n",
    "    buf.write(\"\\\\begin{tabular}{\" + column_format + \"}\\n\")\n",
    "    buf.write(\"\\\\hline\\n\")\n",
    "    \n",
    "    headers = [f\"\\\\textbf{{{latex_escape(col)}}}\" for col in df.columns]\n",
    "    buf.write(\" & \".join(headers) + \" \\\\\\\\\\n\")\n",
    "    buf.write(\"\\\\hline\\n\")\n",
    "\n",
    "    grouped = df.groupby(group_cols, sort=False, dropna=False)\n",
    "    first_block = True\n",
    "\n",
    "    for group_keys, subdf in grouped:\n",
    "        if not isinstance(group_keys, tuple):\n",
    "            group_keys = (group_keys,)\n",
    "\n",
    "        if not first_block:\n",
    "            buf.write(\"\\\\hline\\n\")\n",
    "        first_block = False\n",
    "\n",
    "        row_idx = 0\n",
    "        for _, row in subdf.iterrows():\n",
    "            row_cells = []\n",
    "            for col in df.columns:\n",
    "                if col in group_cols:\n",
    "\n",
    "                    if row_idx == 0 or subdf.iloc[row_idx - 1][col] != row[col]:\n",
    "                        \n",
    "                        same_count = (subdf[col] == row[col]).sum()\n",
    "                        if same_count > 1:\n",
    "                            row_cells.append(\n",
    "                                f\"\\\\multirow{{{same_count}}}{{*}}{{{latex_escape(row[col])}}}\"\n",
    "                            )\n",
    "                        else:\n",
    "                            row_cells.append(latex_escape(row[col]))\n",
    "                    else:\n",
    "                        row_cells.append(\"\")\n",
    "                else:\n",
    "                    row_cells.append(latex_escape(row[col]))\n",
    "            buf.write(\" & \".join(row_cells) + \" \\\\\\\\\\n\")\n",
    "            row_idx += 1\n",
    "\n",
    "    buf.write(\"\\\\hline\\n\\\\end{tabular}\\n\")\n",
    "    if caption:\n",
    "        buf.write(f\"\\\\caption{{{latex_escape(caption)}}}\\n\")\n",
    "    if label:\n",
    "        buf.write(f\"\\\\label{{{latex_escape(label)}}}\\n\")\n",
    "    buf.write(\"\\\\end{table}\\n\")\n",
    "    return buf.getvalue()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "9d197cef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{table}[ht]\n",
      "\\centering\n",
      "\\begin{tabular}{ccccc}\n",
      "\\hline\n",
      "\\textbf{HPol (bp)} & \\textbf{CLNSIG} & \\textbf{Overlap} & \\textbf{Total} & \\textbf{\\%} \\\\\n",
      "\\hline\n",
      "\\multirow{5}{*}{4-6} & Benign & 48677 & 180246 & 0.2701 \\\\\n",
      " & Likely\\_benign & 261084 & 971105 & 0.2689 \\\\\n",
      " & Likely\\_pathogenic & 15228 & 70856 & 0.2149 \\\\\n",
      " & Pathogenic & 16696 & 82029 & 0.2035 \\\\\n",
      " & Uncertain\\_significance & 458592 & 2059317 & 0.2227 \\\\\n",
      "\\hline\n",
      "\\multirow{5}{*}{7-11} & Benign & 3116 & 180246 & 0.0173 \\\\\n",
      " & Likely\\_benign & 12259 & 971105 & 0.0126 \\\\\n",
      " & Likely\\_pathogenic & 435 & 70856 & 0.0061 \\\\\n",
      " & Pathogenic & 377 & 82029 & 0.0046 \\\\\n",
      " & Uncertain\\_significance & 10981 & 2059317 & 0.0053 \\\\\n",
      "\\hline\n",
      "\\multirow{5}{*}{$>11$} & Benign & 1245 & 180246 & 0.0069 \\\\\n",
      " & Likely\\_benign & 2201 & 971105 & 0.0023 \\\\\n",
      " & Likely\\_pathogenic & 86 & 70856 & 0.0012 \\\\\n",
      " & Pathogenic & 69 & 82029 & 0.0008 \\\\\n",
      " & Uncertain\\_significance & 758 & 2059317 & 0.0004 \\\\\n",
      "\\hline\n",
      "\\multirow{5}{*}{$>20$} & Benign & 291 & 180246 & 0.0016 \\\\\n",
      " & Likely\\_benign & 401 & 971105 & 0.0004 \\\\\n",
      " & Likely\\_pathogenic & 10 & 70856 & 0.0001 \\\\\n",
      " & Pathogenic & 16 & 82029 & 0.0002 \\\\\n",
      " & Uncertain\\_significance & 160 & 2059317 & 0.0001 \\\\\n",
      "\\hline\n",
      "\\end{tabular}\n",
      "\\caption{Overlap of ClinVar SNPs per CLNSIG type with homopolymer regions.}\n",
      "\\label{tab1}\n",
      "\\end{table}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_snp = prepare_df(df, vartype=\"SNP\")\n",
    "\n",
    "latex_code = df_to_latex_multirow(\n",
    "    df_snp,\n",
    "    group_cols=[\"HPol (bp)\"],\n",
    "    caption=\"Overlap of ClinVar SNVs per CLNSIG type with homopolymer regions.\",\n",
    "    label=\"tab1\"\n",
    ")\n",
    "\n",
    "print(latex_code)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "8b0c7a1a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{table}[ht]\n",
      "\\centering\n",
      "\\begin{tabular}{ccccc}\n",
      "\\hline\n",
      "\\textbf{HPol (bp)} & \\textbf{CLNSIG} & \\textbf{Overlap} & \\textbf{Total} & \\textbf{\\%} \\\\\n",
      "\\hline\n",
      "\\multirow{5}{*}{4-6} & Benign & 10957 & 28455 & 0.3851 \\\\\n",
      " & Likely\\_benign & 14298 & 37174 & 0.3846 \\\\\n",
      " & Likely\\_pathogenic & 10959 & 35284 & 0.3106 \\\\\n",
      " & Pathogenic & 28537 & 91686 & 0.3112 \\\\\n",
      " & Uncertain\\_significance & 16926 & 58594 & 0.2889 \\\\\n",
      "\\hline\n",
      "\\multirow{5}{*}{7-11} & Benign & 5410 & 28455 & 0.1901 \\\\\n",
      " & Likely\\_benign & 3404 & 37174 & 0.0916 \\\\\n",
      " & Likely\\_pathogenic & 391 & 35284 & 0.0111 \\\\\n",
      " & Pathogenic & 1443 & 91686 & 0.0157 \\\\\n",
      " & Uncertain\\_significance & 1481 & 58594 & 0.0253 \\\\\n",
      "\\hline\n",
      "\\multirow{5}{*}{$>11$} & Benign & 6049 & 28455 & 0.2126 \\\\\n",
      " & Likely\\_benign & 3469 & 37174 & 0.0933 \\\\\n",
      " & Likely\\_pathogenic & 37 & 35284 & 0.001 \\\\\n",
      " & Pathogenic & 35 & 91686 & 0.0004 \\\\\n",
      " & Uncertain\\_significance & 1184 & 58594 & 0.0202 \\\\\n",
      "\\hline\n",
      "\\multirow{5}{*}{$>20$} & Benign & 1262 & 28455 & 0.0444 \\\\\n",
      " & Likely\\_benign & 911 & 37174 & 0.0245 \\\\\n",
      " & Likely\\_pathogenic & 13 & 35284 & 0.0004 \\\\\n",
      " & Pathogenic & 9 & 91686 & 0.0001 \\\\\n",
      " & Uncertain\\_significance & 342 & 58594 & 0.0058 \\\\\n",
      "\\hline\n",
      "\\end{tabular}\n",
      "\\caption{Overlap of ClinVar INDELs per CLNSIG type with homopolymer regions.}\n",
      "\\label{tab2}\n",
      "\\end{table}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_snp = prepare_df(df, vartype=\"INDEL\")\n",
    "\n",
    "latex_code = df_to_latex_multirow(\n",
    "    df_snp,\n",
    "    group_cols=[\"HPol (bp)\"],\n",
    "    caption=\"Overlap of ClinVar Indels per CLNSIG type with homopolymer regions.\",\n",
    "    label=\"tab2\"\n",
    ")\n",
    "\n",
    "print(latex_code)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "classify-indels-env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
